### PR TITLE
add lib.vars to storm

### PR DESCRIPTION
**Pull request by Cisphyx**
Add LibVars to storm for working with variables

### CODE REVIEW

File: synapse/lib/stormtypes.py [hunk 1]
```diff
@@ -841,6 +842,48 @@ def _reqStr(self, name):
...
+    async def _libVarsGet(self, valu, strip=False):
+        '''
+        Resolve a variable in a storm query
+        '''
+        if strip:
+            valu = valu.lstrip('$')
+
+        if valu not in self.runt.vars:
+            mesg = f'No var with name: {valu}'
+            raise s_exc.StormRuntimeError(mesg=mesg, valu=valu, strip=strip) from None
```
Comment (vEpiphyte) [inline synapse/lib/stormtypes.py]: No need to use `from None` here.

File: synapse/lib/stormtypes.py [hunk 2]
```diff
@@ -841,6 +842,46 @@ def _reqStr(self, name):
...
+
+    async def _libVarsDel(self, valu, strip=False):
+        '''
+        Unset a variable in a storm query.
+        '''
+        if strip:
+            valu = valu.lstrip('$')
+
+        try:
+            del(self.runt.vars[valu])
```
Comment (vEpiphyte) [inline synapse/lib/stormtypes.py]: no need to call del directly - `self.runt.vars.pop(valu, None)` is more consistent across the rest of the codebase

Comment (vEpiphyte) [reply synapse/lib/stormtypes.py]: How do we delete vars from Path objects as well?

File: synapse/lib/stormtypes.py [hunk 3]
```diff
@@ -841,6 +842,46 @@ def _reqStr(self, name):
...
+        except:
+            pass
+
+    async def _libVarsList(self):
+        '''
+        List variables available in a storm query.
+        '''
+        return ((key, valu) for key, valu in self.runt.vars.items())
```
Comment (vEpiphyte) [inline synapse/lib/stormtypes.py]: `return list(self.runt.vars.items())`

File: synapse/lib/stormtypes.py [hunk 4]
```diff
@@ -841,6 +842,46 @@ def _reqStr(self, name):
                 ret.append((key, valu))
         return ret
 
+class LibVars(Lib):
+
+    def addLibFuncs(self):
+        self.locls.update({
+            'get': self._libVarsGet,
+            'del': self._libVarsDel,
```
[the remark this answers is not in the document]
Comment (Cisphyx) [reply synapse/lib/stormtypes.py]: Set function has been added

File: synapse/lib/stormtypes.py [hunk 5]
```diff
@@ -998,6 +1050,44 @@ def __init__(self, node, path=None):
...
+        ret = self.path.getVar(name)
+        if ret is s_common.novalu:
+            mesg = f'No var with name: {name}'
+            raise s_exc.StormRuntimeError(mesg=mesg, name=name, strip=strip)
+
+        return ret
+
+    async def _methPathSetVar(self, name, valu, strip=False):
+        '''
+        Resolve a variable in the path of a storm query
```
Comment (vEpiphyte) [inline synapse/lib/stormtypes.py]: docstring

File: synapse/lib/stormtypes.py [hunk 6]
```diff
@@ -841,6 +841,54 @@ def _reqStr(self, name):
...
+        ret = self.runt.getVar(name)
+        if not ret:
+            mesg = f'No var with name: {name}'
+            raise s_exc.StormRuntimeError(mesg=mesg, name=name, strip=strip)
+
+        return ret
+
+    async def _libVarsSet(self, name, valu, strip=False):
+        '''
+        Resolve a variable in a storm query
```
Comment (vEpiphyte) [inline synapse/lib/stormtypes.py]: docstring

**Pull request merged**